tests/std/tests/P0896R4_views_iota/test.cpp

```diff
@@ -239,6 +241,57 @@ constexpr void test_integral() {
     }
 }
 
+template <class W>
+constexpr void test_one_difference() {
+    ranges::iota_view r{numeric_limits<W>::min(), numeric_limits<W>::max()};
+    using Diff = conditional_t<sizeof(W) >= sizeof(long long), _Signed128,
+        conditional_t<sizeof(W) >= sizeof(int), long long, int>>;
+    static_assert(same_as<decltype(r.end() - r.begin()), Diff>);
+    static_assert(noexcept(r.end() - r.begin()));
+    const auto n = (Diff{1} << (numeric_limits<W>::digits + (signed_integral<W> ? 1 : 0))) - 1;
+    assert(r.end() - r.begin() == n); // left > right
+    assert(r.begin() - r.end() == -n); // right > left
+}
+
+constexpr bool test_difference() {
+    // Ensure we have full coverage of all branches in `operator-(i, s)`
+
+    // signed integer-like, sizeof(W) < sizeof(int)
+    test_one_difference<signed char>();
+    test_one_difference<short>();
+
+    // unsigned integer-like, sizeof(W) < sizeof(int)
+    test_one_difference<unsigned char>();
+    test_one_difference<unsigned short>();
+
+    // signed integer-like, sizeof<int> <= sizeof(W) < sizeof(long long)
+    test_one_difference<int>();
+
+    // unsigned integer-like, sizeof<int> <= sizeof(W) < sizeof(long long)
+    test_one_difference<unsigned int>();
+
+    // signed integer-like, sizeof<long long> <= sizeof(W)
+    test_one_difference<long long>();
+
+    // unsigned integer-like, sizeof<long long> <= sizeof(W)
+    test_one_difference<unsigned long long>();
+
+    // non-integer-like
+    {
+        char* some_chars = new char[1ull << 24];
+        ranges::iota_view r{&some_chars[0], &some_chars[1ull << 24]};

```

Comment:
Per CWG-232, `&*nullptr` and `&array[array_size]` (by virtue of being equivalent to `&*(array + array_size)`) are all well-formed since they do not perform lvalue-to-rvalue conversion on the so-called "empty lvalue" that results from the dereference. The issue has been in "Drafting" for at least 17 years, but CWG's intent has been consistent since it was filed in June 2000. This is sufficient for my bar in personal projects, and I maybe even would have used it in the pre-open source STL, but today I don't want readers wasting their time wondering if this is a mistake.